paddlenlp/transformers/t5/modeling.py

```diff
@@ -91,13 +91,14 @@ def __init__(self, hidden_size, eps=1e-6):
 
     def forward(self, hidden_states):
         # layer norm should always be calculated in float32
-        variance = paddle.pow(hidden_states.astype(paddle.float32), 2).mean(axis=-1, keepdim=True)
-        hidden_states = hidden_states * paddle.rsqrt(variance + self.variance_epsilon)
-
-        # convert into float16 if necessary
-        if amp_state() or self.weight.dtype == paddle.float16:
-            hidden_states = hidden_states.astype(paddle.float16)
-        return self.weight * hidden_states
+        with paddle.amp.auto_cast(enable=False):
+            variance = paddle.pow(hidden_states.astype(paddle.float32), 2).mean(axis=-1, keepdim=True)
+            hidden_states = hidden_states * paddle.rsqrt(variance + self.variance_epsilon)
+
+            # convert into float16 if necessary
+            if amp_state() or self.weight.dtype == paddle.float16:
+                hidden_states = hidden_states.astype(paddle.float16)
+            return self.weight * hidden_states
```

Comment:
```suggestion
        variance = paddle.pow(hidden_states, 2).mean(axis=-1, keepdim=True)
        hidden_states = hidden_states * paddle.rsqrt(variance + self.variance_epsilon)

        with paddle.amp.auto_cast(self.weight.dtype == paddle.float16):
            result = self.weight * hidden_states
        return result
```

你在 auto_cast 里面将enable=False， 强制设置成 false，就没办法实现混合计算，会导致精度损失的问题。

什么时候需要`auto_cast`： 当 self.weight.dtype == paddle.float16 的时候才需要，所以用这个就行了。